scripts/modules/b2b-account/account-hierarchy.js

```diff
@@ -0,0 +1,140 @@
+                $(".account-hierarchy>ul.tree li").first().children(".nested").addClass("active");
+            });
+        },
+        toggleNodes: function (e) {
+            $(e.currentTarget).parent().parent().children(".nested").toggleClass("active");
+            $(e.currentTarget).toggleClass("caret-down");
+        },
+        showThreeDotsMenu: function (e) {
+            $(e.currentTarget).closest('li').children('.dropdown-content').toggleClass("active");
+        },
+        addChildAccount: function (e) {
+            //todo: need to implement this method in future.
+        },
+        expandAll: function (e) {
+            $(".tree .caret").addClass("caret-down");
+            $(".nested").addClass("active");
+        },
+        collapseAll: function (e) {
+            $(".tree .caret").removeClass("caret-down");
+            $(".nested").removeClass("active");
+        },
+        viewAccount: function (e) {
+            //todo: need to implement this method in future.
+            var accountId = e.currentTarget.dataset.mzValue;
+        },
+        changeParentAccount: function (e) {
+            //todo: need to implement this method in future.
+            var accountId = e.currentTarget.dataset.mzValue;
+        }
+    });
+
+    var AccountHierarchyModel = Backbone.MozuModel.extend({
+        mozuType: 'accounthierarchy',
+        initialize: function () {
+            var self = this;
+            self.initApiModel();
+            self.apiGet({ id: require.mozuData('user').accountId }).then(function (response) {
+                self.set("hierarchy", self.processAccountHierarchy(response.data));
+                self.syncApiModel();
+            });
+            self.set("isUserAdmin", self.isUserAdmin());
+        },
+        isUserHavingBehavior: function (behaviorId) {
+            var behaviors = require.mozuData('user').behaviors;
+            if (behaviors) {
+                return behaviors.includes(behaviorId);
+            }
+            return false;
+        },
+        isUserAdmin: function () {
+            //Only admin should have 1000 behavior.
+            return this.isUserHavingBehavior(1000);
+        },
+        isUserPurchaser: function () {
+            //purchaser have 1005 behavior
+            return !this.isUserAdmin() && this.isUserHavingBehavior(1005);
+        },
+        processAccountHierarchy: function (data) {
+            var self = this;
+            if (data.hierarchy) {
+                self.setAccountOnHierarchy(data.hierarchy, data.accounts);
+            }
+            return data.hierarchy;
+        },
+        setAccountOnHierarchy: function (item, accounts, isCurrentAccountChildrens) {
+            var self = this;
+            // isCurrentAccountChildrens will be null/false initially but when current account is logged in users account then set it to true. so we can use this to show menu on children.
```

Comment:
Comment is good but grammar. I'd recommend something like below. Food for thought.
```
// This parameter will be false until recursion finds the account associated
// with the current user. On further recursion it will enable descendant behaviors.
```